docs/Cluster-Internal-networks.md

```diff
@@ -0,0 +1,42 @@
+- Storage Network - connecting NAS storage (NetApp) to each of the hosts within a cluster.
+- Pod Network - IP addresses assigned to each pod.
+- Service Network - IP addresses assigned to services.
+
+## Network usage
+
+### Host/Node Network
+
+Host IP addresses are used to NAT each hosted PODs outgoing traffic, as well as host specific node/cluster related services.  Incoming application traffic will not be directed at a host IP address.  While we are planning to use outgoing NAT on the F5, there are situations where we may want to be able to direct route to these host IP addresses from inside the data-centres.  Current OCP3 leverages public IP addresses for all host/nodes so a pre-build list of services that would require direct access has not been described/tested yet.
+
+A separate (unique) Host/Node network for each cluster is recommended.
+
+### Storage Network
+
+This network will be leveraged for high speed communication between the NetApp appliance and the cluster Hosts/Nodes.  While no other access is planned for this network, there are still design elements to identify (eg: backup network integration, etc) that may benefit from a unique network assignment.
+
+A separate (unique) Storage network for each cluster is recommended.
+
+### Cluster Pod and Service Networks
+
+These networks are used by internal pod-pod communication while external cluster access (both incoming and outgoing) is routed through the host/node IP address or a routed ingress IP.  This makes these networks good candidates for re-using network ranges.
+
+A known technology that will be impacted by this decision is the Istio Service Mesh implementation.  The internal pod and service networks are used for service discovery within a Mesh backplane.  One design for a multi-cluster deployment uses a single (shared) Istio backplane.  This design requires uniquely identifiable pod endpoints.  A second design uses separate Istio backplanes for each cluster, and replication between backplanes provides cross-cluster service discovery.  This design would NOT require uniquely identifiable pod endpoints between clusters.
+
+Choosing to re-use the internal cluster networks (for pods and services) will remove the option to stretch an Istio Backplane across multiple clusters.  Requesting additional private networks for each production cluster will require requests for reserved network ranges from hosting services.
+
+The current production architecture defines a *Silver* platform that includes a single cluster, and *Gold* platform that is made up of 2 clusters in separate physical regions.
+
+The recommendation is to leverage unique cluster pod/service networks within each region.  This would allow the option of a single Istio Backplane across the Gold clusters in the future, while not demanding unique pod/service networks in every cluster.  This would allow partial re-use of reserved networks while not blocking a future design pattern we may need to leverage.
```

Comment:
I think we should also cover how big of a network this needs to be.

https://docs.openshift.com/container-platform/4.4/installing/installing_bare_metal/installing-bare-metal.html#installation-bare-metal-config-yaml_installing-bare-metal

`hostPrefix` determines how many pods you can have on a host. This can't be changed later. Red Hat recommends 23 to allow for 512 pods per host as is allowed in OCP 4.4.

`hostPrefix` times the maximum number of nodes in the cluster will give the full CIDR size needed.

It's my understanding that both `clusterNetwork` and `serviceNetwork` should be the same size.

Reply:
```suggestion
The recommendation is to leverage private non-routable cluster pod/service networks within each region.  This would allow the possibility for a single Istio Backplane across the Gold clusters in the future, while not demanding separate non-routable pod/service networks in every cluster.  This would allow partial re-use of cluster internal networks while not blocking a future design pattern we may need to leverage.
```